File: src/api/RiskAnalytics.Api.Model/SavedSearchModel.cs
namespace RiskAnalytics.Api.Model;

public class SavedSearchModel: AssetWatchTableSearchParameters
{
    public int Id { get; set; }
    public int PortfolioId { get; set; }
    public string PortfolioName { get; set; }
    public string Name { get; set; } = null!;
    public string? Description { get; set; }
    public bool IsActive { get; set; }
    public string UserId { get; set; } = null!;
    public DateTime DateCreated { get; set; }
    public DateTime DateModified { get; set; }
}

File: src/frontend/src/app/modules/asset-watch/models/saved-search-model.ts
import { AssetWatchGridRequest } from './asset-watch-grid-request';

export class SavedSearchModel extends AssetWatchGridRequest {
  id!: number;
  name!: string;
  portfolioName!: string;
  description?: string;
  isActive!: boolean;
  userId!: string;
  dateCreated!: Date;
  dateModified!: Date;
}
